.mergify.yml

```diff
@@ -9,4 +9,15 @@ pull_request_rules:
       merge:
         method: squash
         strict: smart+fasttrack
-        commit_message: title+body
\ No newline at end of file
+        commit_message: title+body
+  - name: automatic merge for Dependabot pull requests
+    conditions:
+      - author~=^dependabot(|-preview)\[bot\]$
```

Comment:
If I read this correctly this allows these accounts: `dependabot[bot]`, `dependabot-preview[bot]`. Is the `[bot]` really part of the author field?

Reply:
It looks like the author field is `dependabot-preview[bot]`:
```
$ git log 9b9189ff755ebe3bca0878caff9d1183e4eed790 -1  | grep Author
Author: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>
```

Reply:
But if this is the git author field then anyone can set their name to this right? I thought this was referring to the github account, otherwise it doesn't protect against malicious PRs.

Reply:
> But if this is the git author field then anyone can set their name to this right?

Hmm, good point. Maybe we can check the commit GPG signature, since `dependabot` uses signed commits.

Reply:
I created https://github.com/gnosis/dex-services/pull/1125/commits and simulated it on `mergify.io` with the given config:

<img width="1236" alt="Screen Shot 2020-07-14 at 16 41 36" src="https://user-images.githubusercontent.com/1200333/87439313-eac7b180-c5f0-11ea-8ed7-92a5099c75c2.png">

While the author of a commit can be forged easily the author of a pull request seems to have to be an authenticated github user. As such the dependabot-bot username needs to be unique.

> Maybe we can check the commit GPG signature, since dependabot uses signed commits.

I don't think we can make mergify rules based on signatures (couldn't find it here: https://doc.mergify.io/conditions.html)

Reply:
> author of a pull request

Oh yea! that makes a lot of sense...

thanks for looking into this.